design/logging.md

```diff
@@ -0,0 +1,542 @@
+ INFO 2020/05/17 17:39:58 +0000 [pid=401] [origin=127.0.0.1] [request_id=f39fb7e2-59f9-4f3a-89a9-bb9a50b09860] [tid=413] CONJ00017I Fetched Identity Provider keys from cache successfully
+ INFO 2020/05/17 17:39:58 +0000 [pid=401] [origin=127.0.0.1] [request_id=f39fb7e2-59f9-4f3a-89a9-bb9a50b09860] [tid=413] CONJ00005I Token decoded successfully
+ INFO 2020/05/17 17:39:58 +0000 [pid=401] [origin=127.0.0.1] [request_id=f39fb7e2-59f9-4f3a-89a9-bb9a50b09860] [tid=413] CONJ00004I Extracted username 'not_in_conjur' from ID token field 'preferred_username'
+ INFO 2020/05/17 17:39:59 +0000 [pid=401] [origin=127.0.0.1] [request_id=f39fb7e2-59f9-4f3a-89a9-bb9a50b09860] [tid=413] Authentication Error: #<Errors::Authentication::Security::RoleNotFound: CONJ00007E 'not_in_conjur' not found>
+ INFO 2020/05/17 17:39:59 +0000 [pid=401] [origin=127.0.0.1] [request_id=f39fb7e2-59f9-4f3a-89a9-bb9a50b09860] [tid=413] Completed 401 Unauthorized in 143ms
+```
+
+However, this doesn't really break any backwards-compatibility, rather than that 
+it add more messages to the log. 
+
+I don't think that we should change the default log level to `warn` because then 
+we will not get the useful `INFO` logs that we got earlier (and that _will_ be a breaking change). 
+
+I suggest that we stay with `info` as the default log level and add a note in the 
+documentation stating that for a loaded environment it is best to change the log 
+level to `warn` to reduce the log size.  
+
+### Affected Components
+
+- Conjur
+- DAP
+
+## Security
+
+All the messages that were previously written to the logs only when the log level
+was changed to DEBUG, will now be printed by default. Some of these messages include
+errors that come from third-parties so we can't control what's written in them and they
+can potentially include sensitive data.
+
+However, I don't think that changing the log level should add any security concern as the
+Conjur log is visible to the same persona that can change the log level, so it really doesn't 
+give the user any data that wasn't available for them before.
+
+## Test Plan
+
+We will not introduce new tests in this effort and the regression tests will be enough.
+
+## Documentation
+
+In our [current docs](https://docs.cyberark.com/Product-Doc/OnlineHelp/AAM-DAP/Latest/en/Content/Operations/Services/logs.htm#Enabledebuglogmessages)
+we explain how users can enable debug logs. We will need to update that page to explain
+on the differences between INFO and DEBUG.
+
+We should also add a note suggesting to change the log level to `warn` in loaded environments.
+
+Note: The page above exists only for DAP as we have an issue in Conjur where it's impossible 
+to restart the server. Once that is solved we can create the page for Conjur as well.
+
+We will also add an entry to the CHANGELOG of this project.
+
+## Version update
+
+- Conjur
+- DAP
```

Comment:
@jvanderhoof @alexkalish do we publish a version for changes like this or are they accumulated to the next release? same question for Conjur and DAP.

Reply:
It really depends on the need.  If we want to get a change into the hands of a customer, we'll cut a release.